libs/common/src/change-stream/test-helpers/with-delete-many.api-tests.ts

```diff
@@ -0,0 +1,196 @@
+import { AuditDatabaseRecord, DeletionAuditLog, MongoDbCollectionName } from '../../types';
+import { changeStreamConfig } from '../config';
+
+const { CHANGE_STREAM_ENABLED } = changeStreamConfig;
+
+type Params = {
+  makeRequest: () => Promise<void>;
+  collectionName: MongoDbCollectionName;
+  auditRecord: AuditDatabaseRecord;
+  getDeletedDocumentIds: () => ObjectId[];
+};
+
+export const withDeleteManyTests = ({ makeRequest, collectionName, auditRecord, getDeletedDocumentIds }: Params) => {
+  describe(`when deleting a document from ${collectionName}`, () => {
+    let mongoDbClient: MongoDbClient;
+    let deletionAuditLogsCollection: Collection<WithoutId<DeletionAuditLog>>;
+    const deleteManyMock = jest.spyOn(Collection.prototype, 'deleteMany');
+    const insertManyMock = jest.spyOn(Collection.prototype, 'insertMany');
+
+    beforeAll(async () => {
+      mongoDbClient = new MongoDbClient({
+        dbName: process.env.MONGO_INITDB_DATABASE as string,
+        dbConnectionString: process.env.MONGODB_URI as string,
+      });
+
+      deletionAuditLogsCollection = await mongoDbClient.getCollection('deletion-audit-logs');
+    });
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    afterAll(async () => {
+      await mongoDbClient.close();
+      deleteManyMock.mockRestore();
+      insertManyMock.mockRestore();
+    });
+
+    if (CHANGE_STREAM_ENABLED) {
+      describe('when the service is working normally', () => {
+        it('should add a deletion audit log', async () => {
+          await makeRequest();
+
+          const deletionAuditLogs = await deletionAuditLogsCollection
+            .find({
+              $or: getDeletedDocumentIds().map((deletedDocumentId) => ({
+                collectionName: { $eq: collectionName },
+                deletedDocumentId: { $eq: deletedDocumentId },
+              })),
+            })
+            .toArray();
+
+          expect(deletionAuditLogs).toEqual(
+            getDeletedDocumentIds().map((id) => ({
+              _id: expect.any(ObjectId) as ObjectId,
+              collectionName,
+              deletedDocumentId: new ObjectId(id),
+              auditRecord,
+              expireAt: expect.any(Date) as Date,
+            })),
+          );
+        });
+
+        it('should delete the documents', async () => {
+          await makeRequest();
+
+          const collection = await mongoDbClient.getCollection(collectionName);
+          const deletedDocuments = await collection
+            .find({
+              $or: getDeletedDocumentIds().map((deletedDocumentId) => ({
+                _id: { $eq: deletedDocumentId },
+              })),
+            })
+            .toArray();
+
+          expect(deletedDocuments.length).toBe(0);
+        });
+      });
+
+      describe('when deleting the document is not acknowledged', () => {
+        beforeEach(() => {
+          when(deleteManyMock)
+            // @ts-ignore
```

Comment:
It's strange that the `deleteManyMock` doesn't seem to recognise that the callback argument is optional :/ However, given that there doesn't appear to be an easy workaround and the callback argument is being removed in the next version of the MongoDB client anyway, I'm happy to leave this.